src/OpenTelemetry/BatchExportProcessor.cs

```diff
@@ -39,8 +39,8 @@ public abstract class BatchExportProcessor<T> : BaseExportProcessor<T>
         private readonly int maxExportBatchSize;
         private readonly Thread exporterThread;
         private readonly AutoResetEvent exportTrigger = new AutoResetEvent(false);
-        private readonly ManualResetEvent dataExportedNotification = new ManualResetEvent(false);
-        private readonly ManualResetEvent shutdownTrigger = new ManualResetEvent(false);
+        private readonly ManualResetEventSlim dataExportedNotification = new ManualResetEventSlim(false);
```

Comment:
Just to provide more context here so I don't forget about it again:

1. When we expect a computational intensive situation, by just spinning/burning few CPU cycles we're very likely to get unblocked, the slim version will be beneficial since the CPU cycles burnt by the spin is expected to be less than the cycles needed to do a CSwitch.
2. Another situation where spin could be useful, is to avoid giving up the quorum. For example, when measuring the duration of an operation, we don't want the instrumentation code to cause extra CSwitch - it is just too much [observer effect](https://en.wikipedia.org/wiki/Observer_effect_(physics)).

In addition, spin makes no sense for if the process only has access to one CPU core.

Given the exporter would typically involve I/O, having the hot path (user threads that are sending telemetry data to the batch processor) spinning won't help.
Given the exporter is running on its own background thread, it would be rare to have requirement on 2).